### PR TITLE
Change hash to array in tome.pm

### PR DESCRIPTION
**Pull request by csepanda**
Replace the hash tome with an array, update corresponding subroutines. Fix documentation(Copy-Paste is an evil).

### CODE REVIEW

File: src/tome.pm [hunk 1]
```diff
@@ -108,9 +108,11 @@ sub read_tome_file($$) {
 
    if (-r $tome_file) {
       open my $tome_fh, "<:utf8", $tome_file or warn "Can't open $tome_file!";
-      chomp, $tome{$_} = 1 while (<$tome_fh>);
+      @tome = <$tome_fh>;
```
Comment (zhmylove) [inline src/tome.pm]: chomp(@tome = <$tome_fh>);

File: src/tome.pm [hunk 2]
```diff
@@ -108,9 +108,11 @@ sub read_tome_file($$) {
...
+      @tome = grep { !/^[\s\d-+]+$/ } @tome;
+      chomp @tome;
```
Comment (zhmylove) [inline src/tome.pm]: It's not needed at all ;)

File: src/tome.pm [hunk 3]
```diff
@@ -108,9 +108,11 @@ sub read_tome_file($$) {
 
    if (-r $tome_file) {
       open my $tome_fh, "<:utf8", $tome_file or warn "Can't open $tome_file!";
-      chomp, $tome{$_} = 1 while (<$tome_fh>);
+      @tome = <$tome_fh>;
+      @tome = grep { !/^[\s\d-+]+$/ } @tome;
+      chomp @tome;      
       close $tome_fh;
-      say "Tome records: " . keys %tome if scalar keys %tome;
+      say "Tome records: " . scalar @tome if scalar @tome;
```
Comment (zhmylove) [inline src/tome.pm]:
```perl
if scalar @tome;
```
Why did you use explicit scalar context? We can achieve the same results with:
```perl
if @tome;
```

File: src/tome.pm [hunk 4]
```diff
@@ -144,18 +146,19 @@ saved in the array previously and uses it as a return value.
...
-   my $txt = shift // die 'Insufficient arguments';
-
-   my $rndkey = (keys %tome)[rand keys %tome];
+   my $txt  = shift // die 'Insufficient arguments';
    
-   if ($txt =~ m{[^\s\n+\d-]}) {
+   my $rnd_idx = int rand scalar @tome; 
+   my $rnd_msg = $tome[$rnd_idx];
+   
+   if ($txt =~ m{[^\s+\d-]}) {
```
Comment (zhmylove) [inline src/tome.pm]: Change code style to write the [] in the same manner as above: `[^\s\d-+]`

File: src/tome.pm [hunk 5]
```diff
@@ -144,18 +146,19 @@ saved in the array previously and uses it as a return value.
...
       my $txt = (split '\n', $txt)[0];
-
-      delete $tome{ $rndkey } if (keys %tome >= $tome_max);
-      $tome{$txt} = substr $txt, 0, $tome_msg_max;
+      $txt = substr ($txt, 0, $tome_msg_max) if length $txt >= $tome_msg_max;
```
Comment (zhmylove) [inline src/tome.pm]: What for it that if:
```perl
if length $txt >= $tome_msg_max;
```
You know for sure that it is needed? May be you've performed some speed tests?

File: src/tome.pm [hunk 6]
```diff
@@ -144,18 +146,19 @@ saved in the array previously and uses it as a return value.
...
+      delete $tome[$rnd_idx] if scalar @tome >= $tome_max;
```
Comment (zhmylove) [inline src/tome.pm]: A deletion from an array is a very expensive operation. 
As I mentioned sometime ago the performant way is to write it like this:
```perl
if (@tome >= $tome_max) {
   $tome[$rnd_idx] = $txt;
} else {
   push @tome, $txt;
}
```

**Review by zhmylove — CHANGES_REQUESTED** [review]
Perfect work!
I left some comments, please change your pull request according to them.
And the last thing - change the commit message name to more descriptive (but not as descriptive as a commit description itself). For example: "Change hash to array in tome.pm"

**Comment by csepanda** [comment]
Re-open an accidentally-closed pull request.

**Review by zhmylove — APPROVED** [review]

File: src/tome.pm (hunk 4 above)
Comment (zhmylove) [reply src/tome.pm]: Lol, morning mistakes: `[^+\s\d-]` of course!

**Pull request merged**